webapp/src/components/AuctionPage/AuctionPage.js

```diff
@@ -25,6 +28,25 @@ export default class AuctionPage extends React.PureComponent {
     }
   }
 
+  componentWillMount() {
+    this.showAuctionModal(this.props)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.showAuctionModal(nextProps)
+  }
+
+  showAuctionModal(props) {
+    const { authorization, onShowAuctionModal } = props
+
+    if (
+      !hasSeenAuctionModal() ||
+      (authorization && !isAuthorized(authorization))
```

Comment:
What do you think of moving`(authorization && !isAuthorized(authorization))` into a variable like

```javascript
const isAuctionContractAuthorized = authorization && !isAuthorized(authorization)
```